### PR TITLE
feat: render connect functions for defined AsyncAPI servers

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR ensures that connection functions for any defined servers in the AsyncAPI document are generated. These are just simple connect function that does not take into account server security etc. That should be solved in a separate PR.

**Related issue(s)**
fixes https://github.com/asyncapi/ts-nats-template/issues/28

### CODE REVIEW

File: components/index/standard.js [hunk 1]
```diff
@@ -141,6 +139,15 @@ function getIsClosedFunction(asyncapi) {
...
 export function getStandardClassCode(asyncapi) {
+  const serverWrapperFunctions = [];
+  for (const [serverName, server] of Object.entries(asyncapi.servers())) {
+    serverWrapperFunctions.push(`
+/**
+ * Connects the client to the AsyncAPI server called ${serverName}.
+ * ${server.description() || ''}
+ */
+async connectTo${pascalCase(serverName)}(){ await this.connect({ servers: ["${server.url()}"] }); }`);
+  }
```
Comment (magicmatatjahu) [inline components/index/standard.js:150]: This logic needs separate function :) Overall good, except that.

File: components/index/standard.js [hunk 2]
```diff
@@ -264,7 +269,7 @@ async connectTo${pascalCase(serverName)}(){ await this.connect({ servers: ["${se
       });
     }
     
-    ${serverWrapperFunctions.join('\n')}`;
+    ${renderConnectServerFunctions(Object.entries(asyncapi.servers())).join('\n')}`;
```
Comment (magicmatatjahu) [inline components/index/standard.js:272]: For me better is passing the object (dictionary with servers and their names), rather than array of arrays.

Comment (magicmatatjahu) [reply components/index/standard.js:272]:
```suggestion
    ${renderConnectServerFunctions(asyncapi.servers()).join('\n')}`;
```

and then in the function:

```ts
for (const [serverName, server] of Object.entries(servers || {})) {
```

Comment (jonaslagoni) [reply components/index/standard.js:272]: You are right, that's better 👍

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**